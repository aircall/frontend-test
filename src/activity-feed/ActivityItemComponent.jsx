import React from "react";
import { withRouter } from "react-router-dom";

import CallMissedOutgoingIcon from "@material-ui/icons/CallMissedOutgoing";
import CallMissedIcon from "@material-ui/icons/CallMissed";
import CallMadeIcon from "@material-ui/icons/CallMade";
import CallReceivedIcon from "@material-ui/icons/CallReceived";
import NavigateNextIcon from "@material-ui/icons/NavigateNext";

export const ActivityItemComponent = props => {
  const { activity, history } = props;

  return (
    <div className="activity-feed-item">
      <div className="item-left">{getIcon(activity)}</div>
      <div className="item-center">
        <div className="activity-feed-item-number">
          {activity.direction === "inbound" ? activity.from : activity.to}
        </div>
        <div className="activity-feed-item-info">{getInfoText(activity)}</div>
      </div>
      <div className="item-right">
        <a onClick={e => goToCallDetail(e, activity.id, history)}>
          <NavigateNextIcon />
        </a>
      </div>
    </div>
  );
};

const getInfoText = activity => {
  const direction = activity.direction === "inbound" ? "on" : "from";
  const callStatus =
    activity.call_type === "answered" ? "called" : "tried to call";

  return `${callStatus} ${direction} ${activity.via}`;
};

const getIcon = activity => {
  if (activity.direction === "inbound") {
    return activity.call_type === "answered" ? (
      <CallReceivedIcon />
    ) : (
      <CallMissedIcon />
    );
  } else {
    return activity.call_type === "answered" ? (
      <CallMadeIcon />
    ) : (
      <CallMissedOutgoingIcon />
    );
  }
};

const goToCallDetail = (e, activityId, history) => {
  e.preventDefault();
  const newLocation = `/call-detail/${activityId}`;
  history.push(newLocation);
};

export default withRouter(ActivityItemComponent);
