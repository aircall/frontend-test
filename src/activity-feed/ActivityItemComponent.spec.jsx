import React from "react";
import renderer from "react-test-renderer";
import { ActivityItemComponent } from "./ActivityItemComponent";
describe("ActivityItemComponent", () => {
  test("snapshot renders", () => {
    const activity = {
      id: 7834,
      created_at: "2018-04-19T09:38:41.000Z",
      direction: "outbound",
      from: "Pierre-Baptiste BÃ©chu",
      to: "06 46 62 12 33",
      via: "NYC Office",
      duration: "120",
      is_archived: false,
      call_type: "missed"
    };
    const component = renderer.create(
      <ActivityItemComponent activity={activity} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
