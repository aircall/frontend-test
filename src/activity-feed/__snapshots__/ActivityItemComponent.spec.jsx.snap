// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityItemComponent snapshot renders 1`] = `
<div
  className="activity-feed-item"
>
  <div
    className="item-left"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <defs>
        <path
          d="M24 24H0V0h24v24z"
          id="a"
        />
      </defs>
      <path
        d="M3 8.41l9 9 7-7V15h2V7h-8v2h4.59L12 14.59 4.41 7 3 8.41z"
      />
    </svg>
  </div>
  <div
    className="item-center"
  >
    <div
      className="activity-feed-item-number"
    >
      06 46 62 12 33
    </div>
    <div
      className="activity-feed-item-info"
    >
      tried to call from NYC Office
    </div>
  </div>
  <div
    className="item-right"
  >
    <a
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
      </svg>
    </a>
  </div>
</div>
`;
